api/src/extensions/manager.ts

```diff
@@ -220,17 +221,20 @@ export class ExtensionManager {
 	 */
 	public async install(versionId: string): Promise<void> {
 		await this.installationManager.install(versionId);
-
-		// Publish a message so all instances will reload extensions. Note: this will also reload the
-		// current instance, as it's subscribed to the same channel
 		await this.messenger.publish(this.reloadChannel, {});
+
+		// This is an incredibly dirty hack to work around the fact that we currently don't have a
+		// reliable way to wait for the listening containers to be done reloading
+		await setTimeout(1000);
```

Comment:
As you mention in your comment this will of course fail for people on slow networks or downloading large extension bundles or both.